src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -559,6 +565,62 @@ def private_numbers(self):
             )
         )
 
+    def as_bytes(self, encoding, format, encryption_algorithm):
+        if not isinstance(encoding, Encoding):
+            raise TypeError("encoding must be an item from the Encoding enum")
```

Comment:
Experience has taught me that any error message like this *really* ought to have an "… instead got …" portion which explains what object was received instead.  Otherwise debugging can be agonizing - "I only get this exception in production, my unit tests aren't catching it, what am I doing? am I reversing the order of arguments, is something naming a variable wrong..."